lib/lrmd/lrmd_client.c

```diff
@@ -1404,7 +1469,7 @@ add_tls_to_mainloop(lrmd_t *lrmd, bool do_handshake)
                                    native->server, native->port);
 
     struct mainloop_fd_callbacks tls_fd_callbacks = {
-        .dispatch = lrmd_tls_dispatch,
+        .dispatch = lrmd_tls_dispatch_async,
```

Comment:
The notify trigger just after this (which reuses `lrmd_tls_dispatch()` as its handler) is interesting.

`lrmd_server_send_notify()` sends messages with type "notify", and `read_remote_reply()` appends those to the `pending_notify` list and sets the trigger (if they arrive while trying to read a reply).

Using `lrmd_tls_dispatch()` as the trigger handler means it will process the `pending_notify` list, and then try to read another message. (And that read will be blocking. With this current PR, `lrmd_tls_dispatch_async()` will never see any pending notifies since it's not set as the trigger handler.)

It seems to me that something like `process_pending_notifies()` should be the trigger handler. I don't know why we'd want to try to read another message at that point; the dispatch function should be called separately if there's any data on the socket.

The trigger handler should return `G_SOURCE_CONTINUE` (TRUE) or `G_SOURCE_REMOVE` (FALSE), but since we're currently overloading `lrmd_tls_dispatch()` as the handler, we fudge that. The socket handler should return a negative value to drop the connection or anything else to continue. Presumably returning 1 or -1 works the same as `G_SOURCE_CONTINUE`.

Reply:
Updated to address these issues, but it doesn't seem to fix the problem.  I'm not exactly sure on the return values here, but I think I got them right.